accepted/00000-rbac.md

```diff
@@ -0,0 +1,285 @@
+
+- Activation Key Administrator
+- Configuration Administrator
+- Image Administrator
+- Channel Administrator
+- System Group Administrator
+- Read-only API User
+
+Each of these roles grants access to different features in Uyuni. This static access control is enforced individually within each feature. Stripping these control checks throughout the system requires significant effort. Therefore, in the initial phase, the logic of these static roles will coexist with their RBAC counterparts, but will be bypassed by assigning them to all existing and future users. This will effectively make RBAC the exclusive access control mechanism.
+
+
+## 2. User Access Information
+
+Users can be granted access either individually or through access groups.
+
+Users assigned to an access group inherit all permissions defined within that group.
+While administrators can grant individual users additional permissions beyond those provided by their assigned group, they cannot revoke permissions set by the group. Additionally, users can belong to multiple access groups and inherit all associated access rules.
+
+Predefined access groups are designed to meet the needs of most environments; however, administrators can create custom groups to address specific requirements as needed.
+
+Below is a conceptual example of how access rules can be defined for an individual user:
+
+**Example:** Grant tailored access to content management (images) feature for user `Alice`
+
+Available namespaces in the content management feature:
+
+|    namespace      | access_mode |                               description
+|-------------------|-------------|-------------------------------------------------------------------------
+|cm.build           | Modify (W)  | Build container or Kiwi images
+|cm.image.import    | Modify (W)  | Import container images from a registered image store
+|cm.image.list      | View (R)    | List all images
+|cm.image.list      | Modify (W)  | Delete images
+|cm.image.overview  | View (R)    | View image details, patches, packages, build log and cluster information
+|cm.image.overview  | Modify (W)  | Inspect, rebuild, delete images
+|cm.profile.details | View (R)    | View details of an image profile
+|cm.profile.details | Modify (W)  | Create image profiles, edit profile details
+|cm.profile.list    | View (R)    | List all image profiles
+|cm.profile.list    | Modify (W)  | Delete image profiles
+|cm.store.details   | View (R)    | View details of an image store
+|cm.store.details   | Modify (W)  | Create image stores, edit store details
+|cm.store.list      | View (R)    | List all image stores
+|cm.store.list      | Modify (W)  | Delete image stores
+
+Setting up access rules for user `Alice`:
+
+- Grant view access to the entire images feature:
+
+  ```
+  Grant 'View' on 'cm.*' to 'Alice'
+  ```
+
+- Prevent access to sensitive image store information:
+
+  ```
+  Revoke 'View' on 'cm.store.details' from 'Alice'
```

Comment:
If Alice got this 'View' access through one accessgroup but then here Alice got that permissions revoked, I expect this would take precedence or?

Reply:
If it's removed from User direct grant but is still present in the group, then Alice still has access to the feature. All endpoints are revoked by default, and we are always adding/grant access in all levels we can configure.